utilities/cli/util.py

```diff
@@ -467,3 +467,57 @@ def list_cmake_dir_options(script_dir: Path, cmake_function: str) -> List[str]:
                     except IndexError:
                         continue
     return sorted(results)
+
+
+def build_holohub_path_mapping(
+    holohub_root: Path,
+    project_data: Optional[dict] = None,
+    build_dir: Optional[Path] = None,
+    data_dir: Optional[Path] = None,
+) -> dict[str, str]:
+    """Build a mapping of HoloHub placeholders to their resolved paths"""
+    if data_dir is None:
+        data_dir = holohub_root / "data"
+
+    path_mapping = {
+        "holohub_root": str(holohub_root),
+        "holohub_data_dir": str(data_dir),
+    }
+    if not project_data:
+        return path_mapping
+    # Add project-specific mappings if project_data is provided
+    app_source_path = project_data.get("source_folder", "")
+    if app_source_path:
+        path_mapping["holohub_app_source"] = str(app_source_path)
+    if build_dir:
+        path_mapping["holohub_bin"] = str(build_dir)
+        if app_source_path:
+            try:
+                app_build_dir = build_dir / Path(app_source_path).relative_to(holohub_root)
+                path_mapping["holohub_app_bin"] = str(app_build_dir)
+            except ValueError:
+                # Handle case where app_source_path is not relative to holohub_root
+                path_mapping["holohub_app_bin"] = str(build_dir)
```

Comment:
Have we validated this outside of the main HoloHub repo, i.e. in a mini-Holohub repo? Can we?

Reply:
not yet, I feel we can validate more when the use case is clearer? currently all the `project_data` that we are dealing with are from this 'find_project' function, which only support the primary repo or a repo in a very similar folder structure... https://github.com/nvidia-holoscan/holohub/blob/5db60cfc20d737c0dbd3863d981624223db5b8ab/utilities/cli/holohub.py#L385

Reply:
Agreed this doesn't need to block the PR. We should validate support in the near future, probably with CI coverage. Opened an internal ticket to track.